src/providers/data_provider/dp_client.c

```diff
@@ -104,6 +104,11 @@ dp_client_register(TALLOC_CTX *mem_ctx,
 
     dp_cli = sbus_connection_get_data(cli_conn, struct dp_client);
 
+    if (dp_cli == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Unknown client! [%s]\n", name);
```

Comment:
`sbus_connection_get_data()` should get the data / payload and it fails. I think this error message should be about this data being broken. Also `EINVAL` may be better fit for the return value. @pbrezina can you advise here?

Reply:
As said in the ticket, fixing it here would backfire so we need to find the root cause instead.